### ISSUE TITLE
Row inclusion formula

### ISSUE THREAD

**Issue by robreuer** [saltcorn/saltcorn#1573]
Hi all,
I'm struggling to get my Row Inclusion Formula in a list view working.
I want to only show a row if a specific term is in the field users.
The simple case 
users==="rb" 
works perfectly. But if I have several users ("rb, ub, tb") I need to check, whether the user is included in the field like:
users.includes("rb")
But in this case I get the error message 
"An error occurred
A report has been logged and a team of bug-squashing squirrels has been dispatched to deal with the situation."

installed version: 0.8.1-rc.3

Any ideas, how to solve the problem?

**Comment by Tacioandrade** [comment]
I have a similar problem here, I have a combobox that is bringing all the information of all users in a table, even though it is marked as Owned. In the list of items it is working perfectly, but when I make a form with information from 3 tables, it ends up fetching all the groups for some reason.

I need to make it only load the information with its own ID, so I thought based on your tip, use the Where of the form to and put something owner===LoggedIn (which seems to be the ID of the logged in user), but it doesn't recognize that term.

If you or anyone in the group knows which variable I can put to get the logged in user ID I appreciate it.

About your problem, it seems that you will have to use some plugin or something, but in the store I didn't see any that do that.

**Comment by admcgrew** [comment]
I had same problem. Was able to get around using || in formula. Try this:
users==="rb"||users==="ub"||users==="tb"

**Comment by EugeneN** [comment]
This is still an issue. The question boils down to the row inclusion formulas based on many to many relations for [list] views. To the best of my understanding it is not possible still. 

I tried to work around this problem by creating a stored calculated filed which updates the original table with an [json] array of aggregated m2m relations, but then the row inclusion javascript formulas seem to not allow for testing for array inclusion etc.

The original question author circumvented this problem by hardcoding the options to be checked against, but this doesn't work for dynamic data or in the general case.

I did briefly check out the expression compiler [packages/saltcorn-data/models/expression.ts](https://github.com/saltcorn/saltcorn/blob/master/packages/saltcorn-data/models/expression.ts) and couldn't find an option how to make it work with array inclusion checks. Maybe I have missed something.

Could someone provide any hints on this?
Thanks!

UPD: found this: https://github.com/saltcorn/saltcorn/issues/2819#issuecomment-2470825894

It might be a solution, but it is very complicated compared to just having a row inclusion formula capable of making inclusion tests, like this:

`aggregated_json_array_filed.contains(user.id)`

where `.contains()` is a standard Javascript method. `.indexOf(...) >= 0` would also work.
